app/validators/date_validator.rb

```diff
@@ -1,9 +1,10 @@
 # @TODO This class can be much simpler once we are not worrying about values being hashes,
 # strings etc.. (once all forms are converted to NullDbForm)
 class DateValidator < ActiveModel::EachValidator
-  def initialize(omit_day: false, in_the_past: nil, **kwargs)
+  def initialize(omit_day: false, in_the_past: nil, within_range: nil, **kwargs)
```

Comment:
This validator is now both a date validator and a date range validator, I wonder if we should have 2 validators instead ?
The 'in_the_past' is also a range - i.e. the range is from the beginning of time until now

We did talk about this at the beginning but chose to go down this route and just accept an 'in_the_past' option.  However, we are now doing ranges as well, it is just making the responsibility of the validator too much.

I think that 

A DateValidator should validate the date - i.e. it is present and correct
A DateRangeValidator should validate the date range - i.e. it is between x date and y date
A DateRelativeValidator that can validate dates relative to today - this is where your 'in_the_past' should live.  Whilst this could be done as part of the date range validator, having a separate one will mean you can customise the messages better for dates relative to today